redirects/models.py

```diff
@@ -133,7 +131,8 @@ def get_staff_delete_url(self):
     @property
     def obj(self):
         """Work out which object we're pointing to"""
-        return first([getattr(self, f.name, None) for f in self.targets()])
+        field_names = [getattr(self, f.name, None) for f in self.targets()]
+        return next(f for f in field_names if f)  # pragma: no branch
```

Comment:
I'm a bit concerned that we have `# pragma: no branch` in production code, especially when that's code that's just been changed. However, it might well be that it's a pragmatic solution and less risky than removing the `if f` (to use this code snippet as an example). I wouldn't want people to think it was okay these lines weren't tested in future though.

I would recommend either removing the conditional (and manually testing that's okay) or adding in an extra unit test so that coverage doesn't complain. How and when to do that is up to you though.

Reply:
I'll merge this now, and opened #4562 for fixing up this issue.

From memory, I _think_ some of the issue here may also be the code branch reached at the end of the iterator. And not just the `if` branch.

Without testing, it's also possible that this case was buried by virtue of `coverage` not recognising it, but it's made explicit now. See [this comment](https://github.com/nedbat/coveragepy/issues/475#issuecomment-697375303).

It's possible there's some debate on how best to solve that, which is maybe best in a new PR.